packages/fxa-auth-server/lib/payments/capability.ts

```diff
@@ -157,32 +154,30 @@ export class CapabilityService {
    * the user always has proper access to the purchased products.
    */
   public async playUpdate(uid: string, purchase: SubscriptionPurchase) {
-    const affectedProductId = (
-      await this.stripeHelper.purchasesToProductIds([purchase])
+    const affectedPriceId = (
+      await this.stripeHelper.purchasesToPriceIds([purchase])
     ).shift();
-    if (!affectedProductId) {
+    if (!affectedPriceId) {
       // Purchase is not mapped to a product id.
       return;
     }
-    const currentProductIds = await this.subscribedProductIds(uid);
-    let priorProductIds;
-    if (currentProductIds.includes(affectedProductId)) {
+    const currentPriceIds = await this.subscribedPriceIds(uid);
+    let priorPriceIds;
+    if (currentPriceIds.includes(affectedPriceId)) {
       // Remove the product id from the prior list for processing to assume that it
       // was previously inactive and ensure we broadcast a change.
-      priorProductIds = currentProductIds.filter(
-        (id) => id !== affectedProductId
-      );
+      priorPriceIds = currentPriceIds.filter((id) => id !== affectedPriceId);
     } else {
       // Its not a current product. Make sure we broadcast that the capabilities
       // associates with this product are gone by adding it to a prior product list.
-      priorProductIds = [...currentProductIds, affectedProductId];
+      priorPriceIds = [...currentPriceIds, affectedPriceId];
     }
```

Comment:
Why not use the same `Set` approach as in `stripeUpdate`; the logic looks the same -- in fact it may be worth promoting to dedupe and avoid divergent code?